draft-thomson-webpush-vapid.md

```diff
@@ -132,211 +134,279 @@ The words "MUST", "MUST NOT", "SHOULD", and "MAY" are used in this document.
-application server would not be required to accept and store cookies, though the
-push service would not be able to offload any state as a result.
 
+An application might use the Web Push API [API] to include this information.
+For example, the API might permit an application to provide a public key as part
+of a new field on the `PushSubscriptionOptions` dictionary.
 
-# Subscription Association
 
-A stable identifier for an application server - such as a public key or a token
-- could also be used to associate a subscription with the application server.
+Editor's Note:
 
-Subscription association reduces the reliance on endpoint secrecy by requiring
-proof of possession to be demonstrated by an application server when requesting
-delivery of a push message.  This provides an additional level of protection
-against leaking of the details of the push subscription.
+: Allowing the inclusion of multiple keys when creating a subscription would
+  allow a subscription to be associated with multiple application servers or
+  application server instances.  This might be more flexible, but it also would
+  require more state to be maintained by the push service for each subscription.
 
-## Amendments to Subscribing for Push Messages
 
-The user agent includes the public key or token of the application server when
-requesting the creation of a subscription.  For example, the Web Push API [API]
-could allow an application to provide a public key as part of a new field on the
-`PushSubscriptionOptions` dictionary.
+## Using Restricted Subscriptions
 
-This token might then be added to the request to create a push subscription.
+When a push subscription has been associated with an application server, the
+request for push message delivery MUST include proof of possession for the
+associated private key or token that was used when creating the push
+subscription.
 
-Allowing the inclusion of multiple keys when creating a subscription would allow
-a subscription to be associated with multiple application servers or application
-server instances.  This would require more state to be maintained by the push
-service for each subscription.
+A push service MUST reject a message that includes invalid credentials with a
+401 (Unauthorized) status code.  Credentials are invalid if:
 
-## Amendments to Requesting Push Message Delivery
+* either the authentication credentials or public key are not included in the
+  request,
 
-When a subscription has been associated with an application server, the request
-for push message delivery MUST include proof of possession for the associated
-private key or token that was used when creating the subscription.  Requests
-that do not contain proof of possession are rejected with a 401 (Unauthorized)
-status code.
+* the signature on the JWT cannot be successfully verified using the included
+  public key,
 
+* the current time is later than the time identified in the "exp" (Expiry)
```

Comment:
I will add a recommendation.  I think that 24h is a reasonable choice, but don't want to presume.